### PR TITLE
[mono] CodeQL fix set #1

### PR DESCRIPTION
**Pull request by kg**
CodeQL flagged various places where we're dereferencing pointers that could be NULL, this PR systematically cleans some of them up via g_assert. (Note that CodeQL doesn't currently recognize g_assert as an assertion... working on that)

### CODE REVIEW

**Review by steveisok — APPROVED** [review]

File: src/mono/mono/mini/aot-compiler.c [hunk 1]
```diff
@@ -15348,6 +15358,8 @@ set_paths (MonoAotCompile *acfg)
 			acfg->asm_fname = g_strdup_printf ("%s.s", acfg->tmpbasename);
 			acfg->llvm_sfile = g_strdup_printf ("%s-llvm.s", acfg->tmpbasename);
 
+			g_assert (acfg->tmpbasename);
```
Comment (lambdageek) [inline src/mono/mono/mini/aot-compiler.c:15361]: wouldn't it be better to assert before `acfg->tmpbasename` is used - ie before the `g_strdup_printf` calls, above?

File: src/mono/mono/mini/aot-compiler.c [hunk 2]
```diff
@@ -15380,6 +15392,8 @@ set_paths (MonoAotCompile *acfg)
 		} else {
 			acfg->tmpbasename = g_build_filename (acfg->aot_opts.temp_path, "temp", (const char*)NULL);
 			acfg->asm_fname = g_strdup_printf ("%s.s", acfg->tmpbasename);
+
+			g_assert (acfg->tmpbasename);
```
Comment (lambdageek) [inline src/mono/mono/mini/aot-compiler.c:15396]: ditto

**Review by lambdageek — APPROVED** [review]
Looks ok, but I wish we could systematically fix these code patterns:

```c
   SomeResult *result = callee(arg1, arg, error);
   mono_error_assert_ok (error); // we should get to know that result != NULL here
```

by adding some kind of annotation to `callee`.

**Comment by kg** [comment]
> Looks ok, but I wish we could systematically fix these code patterns:
> 
> ```c
>    SomeResult *result = callee(arg1, arg, error);
>    mono_error_assert_ok (error); // we should get to know that result != NULL here
> ```
> 
> by adding some kind of annotation to `callee`.

Yeah, I'm not sure whether there is no relevant annotation or if the annotations just aren't documented.

File: src/mono/mono/mini/aot-compiler.c (hunk 2 above)
Comment (kg) [reply src/mono/mono/mini/aot-compiler.c:15396]: Good catch, I'll fix them.

**Review by fanyang-mono — APPROVED** [review]

**Pull request merged**